Source/bmalloc/libpas/src/libpas/bmalloc_heap_inlines.h

```diff
@@ -80,8 +84,11 @@ static PAS_ALWAYS_INLINE void* bmalloc_allocate_auxiliary_inline(pas_primitive_h
 {
     pas_allocation_result result;
     result = bmalloc_allocate_auxiliary_impl_inline_only(heap_ref, size, 1, allocation_mode);
-    if (PAS_LIKELY(result.did_succeed))
+    if (PAS_LIKELY(result.did_succeed)) {
+        if (allocation_mode == pas_non_compact_allocation_mode && MAR_isAddressInQualifyingPage((void*) result.begin))
+            MAR_didAllocate(&marRegistry, (void*) result.begin, size);
```

Comment:
Ditto. So long as it is not inlined, then it must be tail-call. (And returning the input void*).